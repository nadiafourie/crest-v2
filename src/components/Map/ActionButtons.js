/*
Purpose
  Holds all the Action button for the maps

  the actions are one of the following:
    - add area (open or expands the add area menu)
    - export exports the map to png/svg
    - map layers opens the map layer list

Child Components
  - Map-ActionButton.js

Libs
  - Not sure yet

API
  - Not sure yet

State needed
  - map layer list open or not
  - add area open or not NEW state item
  - Not sure yet

Props
  - Not sure yet
*/
