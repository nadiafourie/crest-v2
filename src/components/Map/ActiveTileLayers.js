import React from 'react'
import { useSelector } from 'react-redux'
import { TileLayer } from 'react-leaflet'

export default function ActiveTileLayers() {

    const activeTileURLS = useSelector((state) => state.mapLayerList.activeLayerTiles)
  return (
      activeTileURLS.map(url => <TileLayer url={url}/>)
  )
}
