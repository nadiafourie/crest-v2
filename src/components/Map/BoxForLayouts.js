import * as React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@mui/styles';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

const useStyles = makeStyles((theme) => ({
  contentBox: {
    padding: theme.spacing(1),
    backgroundColor: theme.palette.CRESTGridBackground.dark,
    borderColor: theme.palette.CRESTBorderColor.main,
    borderStyle: 'solid',
    borderWidth: '1px'
  }
}));

export default function BoxForLayouts(props) {
  const classes = useStyles();
  const { children, boxHeight, boxMarginTop } = props;

  return (
    <Box className={classes.contentBox} style={{ height: boxHeight, marginTop: boxMarginTop }}>
      <Typography variant="body" component="div" gutterBottom>
        {children}
      </Typography>
    </Box>
  );
}

BoxForLayouts.propTypes = {
  children: PropTypes.node.isRequired,
  boxHeight: PropTypes.string.isRequired,
  boxMarginTop: PropTypes.string
};
