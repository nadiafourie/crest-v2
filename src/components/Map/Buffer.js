/*
Purpose
  use or don't use buffer in the search zonal stats or hubs intersection
  not sure what else to include yet
  
Child Components
  - Map.js ?

Libs
  - leaflet

API
  - Not sure yet

State needed
  - None

Props
  - Not sure yet
*/