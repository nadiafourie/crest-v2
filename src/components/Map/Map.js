/*
Purpose
  The leaflet map
  will need interactions between map and charts hovers and clicks on shapes open and higlight each other
    shape and chart

Child Components
  - Map-ActionButton.js
  - Map-ActionButtons.js
  - Map-ClearAllState.js
  - Map-DrawArea.js
  - Map-DrawAreaActions.js
  - Map-MapIdentify.js
  - Map-SearchPlaces.js
  - Map-SearchPlacesAction.js
  - Map-ShareMap.js

Libs
  - leaflet

API
  - indentify
  - zonal stats
  - share map

State needed
  - zoom level
  - center
  - basemap
  - region
  - layers on and off
  - all zonal stat GEOJSON so we can draw it
  - probably missed stuff

Props
  - Not sure yet

*/