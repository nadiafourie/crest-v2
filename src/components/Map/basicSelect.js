import React from 'react'

export function BasicSelect(props) {
  return (
    <select defaultValue={props.defaultValue} onChange={props.onChange}>
      {
        props.values.map((element, key) => <option key={key} value={element}>{element}</option>)
      };
    </select>
  );
}