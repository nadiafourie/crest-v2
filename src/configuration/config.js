import { americanSamoaConfig } from './regions/american_samoa';
import { continentalUSConfig } from './regions/continental_us';
import { guamConfig } from './regions/guam';
import { hawaiiConfig } from './regions/hawaii';
import { northernMarianaIslandsConfig } from './regions/northern_mariana_islands';
import { puertoRicoConfig } from './regions/puertoRico';
import { usVirginIslandsConfig } from './regions/usVirginIslands';
import { alaskaConfig } from './regions/alaska';

export const betaIdentifyEndpoint = 'https://api.resilientcoasts.org/beta/identify/';
export const prodIdentifyEndpoint = 'https://api.resilientcoasts.org/prod/identify/';
export const betaShareLinkEndpoint = 'https://api.resilientcoasts.org/beta/share-link';
export const prodShareLinkEndpoint = 'https://api.resilientcoasts.org/prod/share-link';

export const mapConfig = {
  basemaps: {
    'Dark Gray': 'ArcGIS:DarkGray',
    Imagery: 'ArcGIS:Imagery',
    Streets: 'ArcGIS:Streets',
    Topographic: 'ArcGIS:Topographic'
  },
  regions: {
    // eslint-disable-next-line quote-props
    'Alaska': alaskaConfig,
    'American Samoa': americanSamoaConfig,
    'Continental U.S': continentalUSConfig,
    // eslint-disable-next-line quote-props
    'Guam': guamConfig,
    'Hawai\'i': hawaiiConfig,
    'Northern Mariana Islands': northernMarianaIslandsConfig,
    'Puerto Rico': puertoRicoConfig,
    'US Virgin Islands': usVirginIslandsConfig
  }
};
