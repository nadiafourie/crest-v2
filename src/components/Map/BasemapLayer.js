import React, {
  useEffect, useCallback, useRef
} from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { vectorBasemapLayer } from 'esri-leaflet-vector';
import { mapConfig } from '../../configuration/config';

const basemaps = mapConfig.basemaps;

const baseMapSelector = (state) => state.mapProperties.basemap;

export default function BasemapLayer(props) {
  const { map } = props;
  const selectedBasemap = useSelector(baseMapSelector);
  const basemapRef = useRef(null);
  const handleBasemapChange = useCallback((basemapName) => {
    if (basemapRef.current !== null) {
      basemapRef.current.remove(map);
    }
    if (map) {
      const newBasemap = vectorBasemapLayer(basemaps[basemapName], {
        apikey: 'AAPK19aa44a23e4c4a7788b37541444c07denQmRCIZSmoomcyzoK3i3ko37EDOjKQcz9ui0gt3KdWAk6c2N4fPW9jEBF0__3R7o',
        pane: 'mapPane'
      });
      newBasemap.addTo(map);
      basemapRef.current = newBasemap;
    }
  }, [map, basemapRef]);

  useEffect(() => {
    handleBasemapChange(selectedBasemap);
  }, [selectedBasemap, handleBasemapChange]);

  return (
        <div>BasemapLayer</div>
  );
}

BasemapLayer.propTypes = {
  map: PropTypes.object
};
