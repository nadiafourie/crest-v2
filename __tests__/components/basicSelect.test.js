import React from 'react' ;
import { shallow } from 'enzyme';
import { mapConfig } from '../../src/configuration/config';
import { BasicSelect } from '../../src/components/Map/basicSelect';
import configureStore from 'redux-mock-store'

const initialState = {"value": 'Continental U.S.'}
const mockStore = configureStore()

let store

const regions = mapConfig.regions;
const regionSelector = shallow(<BasicSelect defaultValue={'Continental U.S.'} values={regions}/>)

describe('Region', () => {
    beforeEach(() => {
        store = mockStore(initialState)
    })
    it('Should render the region selector correctly', () => {
        expect(regionSelector).toMatchSnapshot() ;
        console.log(regionSelector.props)
    });
});