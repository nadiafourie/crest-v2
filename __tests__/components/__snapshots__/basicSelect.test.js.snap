// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Region Should render the region selector correctly 1`] = `ShallowWrapper {}`;
